korlibs-memory/src/korlibs/memory/ByteArrayBuilder.kt

```diff
@@ -10,7 +9,7 @@ import kotlin.math.*
 public class ByteArrayBuilder(public var data: ByteArray, size: Int = data.size, public val allowGrow: Boolean = true) {
     public constructor(initialCapacity: Int = 4096) : this(ByteArray(initialCapacity), 0)
 
-    private var _size: Int = size
+    var _size: Int = size
```

Comment:
This shouldn't be exposed.

Reply:
This is the problem with the half structure. If I do not expose code like this, I can't make those functions extention functions. This was my problem in the original PR. I need some help with it, because I don't know how the Half stuff works and why.